File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {

        long count = 0; // current streak of continuous zeros
        long total = 0;

        for (int i = 0; i < nums.length; i++)
            if (nums[i] == 0) {
                count++; // extend the streak of zeros
                total += count; // add all subarrays ending at i
                                // e.g. for 3 zeros → [0], [0,0], [0,0,0]
            } else
                // reset streak when encountering a non-zero
                count = 0;

        return total;
    }
}

File: 2432-number-of-zero-filled-subarrays/number-of-zero-filled-subarrays.py
class Solution:
    def zeroFilledSubarray(self, nums: List[int]) -> int:
        count = 0  # current streak of zeros
        total = 0 

        for num in nums:
            if num == 0:
                count += 1        # extend streak
                total += count    # add all subarrays ending here
            else:
                count = 0         # reset streak

        return total